ipmi-config/ipmi-config-category-core-lan6-conf-section.c

```diff
@@ -0,0 +1,808 @@
+
+	  memset (ipv6_addresses_str, '\0', (BMC_MAXIPV6ADDRLEN+1)*255);
+	  if (NULL == inet_ntop(AF_INET6, ipv6_dynamic_ip_address_bytes, ipv6_address_str, BMC_MAXIPV6ADDRLEN))
+	    {
+	      pstdout_fprintf (state_data->pstate,
+			       stderr,
+			       "fiid_obj_get_data: 'address': %s\n",
+			       strerror (errno));
+	      goto cleanup;
+
+	    }
+	  sprintf(ipv6_addresses_str+strlen(ipv6_addresses_str), "%s ", ipv6_address_str);
+	}
+      fiid_obj_destroy (obj_cmd_rs);
+      obj_cmd_rs = NULL;
+  }
+
+  if (ipmi_config_section_update_keyvalue_output (state_data,
+                                                  kv,
+                                                  ipv6_addresses_str) < 0)
+    return (IPMI_CONFIG_ERR_FATAL_ERROR);
+
+  rv = IPMI_CONFIG_ERR_SUCCESS;
+ cleanup:
+  fiid_obj_destroy (obj_cmd_rs);
+  return (rv);
+}
+
+static ipmi_config_err_t
+ipv6_dynamic_addresses_commit (ipmi_config_state_data_t *state_data,
+                              const char *section_name,
+                              const struct ipmi_config_keyvalue *kv)
+{
+  ipmi_config_err_t rv = IPMI_CONFIG_ERR_FATAL_ERROR;
+
+  assert (state_data);
+  assert (section_name);
+  assert (kv);
+
+  /* Dynamic addresses are not writable. */
+  return (rv);
+}
+
+struct ipmi_config_section *
+ipmi_config_core_lan6_conf_section_get (ipmi_config_state_data_t *state_data,
+                                        unsigned int config_flags,
+                                        int channel_index)
+{
+  struct ipmi_config_section *section = NULL;
+  char *section_comment =
+    "In the Lan6_Conf section, typical networking configuration is setup.  "
+    "Most users will choose to set \"Static\" for the \"IP_Address_Source\" "
+    "and set the appropriate \"IP_Address\", \"MAC_Address\", "
+    "\"Subnet_Mask\", etc. for the machine.";
+  char *section_name_base_str = "Lan6_Conf";
+  unsigned int verbose_option_config_flags = 0;
+
+  assert (state_data);
+
+  /* vlan and ipv4 header parameters not checked out by default */
+
+  if (!state_data->prog_data->args->verbose_count)
+    verbose_option_config_flags = IPMI_CONFIG_DO_NOT_CHECKOUT;
+
+  if (!(section = ipmi_config_section_multi_channel_create (state_data,
+                                                            section_name_base_str,
+                                                            section_comment,
+                                                            NULL,
+                                                            NULL,
+                                                            config_flags,
+                                                            channel_index,
+                                                            state_data->lan_channel_numbers,
+                                                            state_data->lan_channel_numbers_count)))
+    goto cleanup;
+
+
+  if (ipmi_config_section_add_key (state_data,
+                                   section,
```

Comment:
I'll drop the read-only variables.  I mostly added them because I wanted to see them, and I can picture an admin wanting to know if he could really do ipv6-only...  is there a better place to add them so that an admin can read them without sending raw ipmi packets?